src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -128,18 +129,26 @@ def find_subscriptions_on_login(self,  # pylint: disable=too-many-arguments
                 subscriptions = subscription_finder.find_from_user_account(
                     username, password, tenant, self._ad_resource_uri)
 
-        if not subscriptions:
-            raise CLIError('No subscriptions found for this account.')
+        if not allow_no_subscriptions and not subscriptions:
+            raise CLIError("No subscriptions were found for '{}'. If this is expected, use "
+                           "'--allow-no-subscriptions' to have a tenant level account".format(
+                               username))
```

Comment:
It is a bit subtle.
`--allow-no-subscriptions` can be used even if you have subscriptions. The story is, `login` will visit every tenants you are member of. Let us say you have `[t1,t2]`, and there is a subscription `s1` in `t1`, but not any in the `t2`. W/o this flag, the login will only build an account for `t1\s1`, which makes sense because almost all commands need to have subscriptions. But for a few commands which doesn't need subscriptions, say those `ad` ones, users have opened the issue saying they like to run them on `t2`. 
I don't like to build up the tenant level accounts by default, because more likely people don't want it. That is particularly true for RBAC scenarios, that it is common people by mistake forgot to create the role assignment, causing subscription access is not built up. I am open to incoming user voices though, and can switch if needed.
Also, let me know if you have suggestions for a better arg name.